libs/dmlf/include/dmlf/collective_learning/client_algorithm.hpp

```diff
@@ -302,42 +324,57 @@ void ClientAlgorithm<TensorType>::Train()
 template <class TensorType>
 void ClientAlgorithm<TensorType>::Test()
 {
-  // If test set is not available we run test on whole training set
   if (dataloader_ptr_->IsModeAvailable(fetch::ml::dataloaders::DataLoaderMode::TEST))
   {
     dataloader_ptr_->SetMode(fetch::ml::dataloaders::DataLoaderMode::TEST);
-  }
-  else
-  {
-    dataloader_ptr_->SetMode(fetch::ml::dataloaders::DataLoaderMode::TRAIN);
-  }
-
-  // Disable random to run model on whole test set
-  dataloader_ptr_->SetRandomMode(false);
-
-  SizeType test_set_size = dataloader_ptr_->Size();
-
-  dataloader_ptr_->Reset();
-  bool is_done_set;
-  auto test_pair = dataloader_ptr_->PrepareBatch(test_set_size, is_done_set);
-  {
-    FETCH_LOCK(model_mutex_);
+    // Disable random to run model on whole test set
+    dataloader_ptr_->SetRandomMode(false);
 
-    // Set inputs and label
-    auto input_data_it = test_pair.second.begin();
-    auto input_name_it = params_.input_names.begin();
+    SizeType test_set_size = dataloader_ptr_->Size();
 
-    while (input_name_it != params_.input_names.end())
+    dataloader_ptr_->Reset();
+    bool is_done_set;
+    auto test_pair = dataloader_ptr_->PrepareBatch(test_set_size, is_done_set);
     {
-      graph_ptr_->SetInput(*input_name_it, *input_data_it);
-      ++input_name_it;
-      ++input_data_it;
+      FETCH_LOCK(model_mutex_);
+
+      // Set inputs and label
+      auto input_data_it = test_pair.second.begin();
+      auto input_name_it = params_.input_names.begin();
+
+      while (input_name_it != params_.input_names.end())
+      {
+        graph_ptr_->SetInput(*input_name_it, *input_data_it);
+        ++input_name_it;
+        ++input_data_it;
+      }
+      graph_ptr_->SetInput(params_.label_name, test_pair.first);
+
+      test_loss_ = *(graph_ptr_->Evaluate(params_.error_name).begin());
+
+      // Accuracy calculation
+      // Todo: will be replaced by accuracy metric once that PR is merged
+      TensorType test_results = graph_ptr_->Evaluate("FullyConnected_2");
+      test_results            = fetch::math::ArgMax(test_results);
+      SizeType total_score{0};
+      auto     tr = test_results.cbegin();
```

Comment:
m.b. "test_result" and "data_point" instead of tr and dp? because (dp == *tr) condition looks not very self-explanatory ))

Reply:
I agree, but the code is only temporary and I'll remove it as soon as my accuracy metric PR gets merged!

Reply:
I changed it anyway :-)